### PR TITLE
Integrate with remote policies

### LINKED ISSUE #4: Add init command
I'd like to have a command that helps me create an initial validation configuration file - prompting for 
- k8s version
- initial framework(s) to configure (i.e. NSA, PSS, etc)

the output would be a monokle-validation.yaml file with the corresponding plugins/rules/settings.

### LINKED ISSUE #14: The 'fw' alias doesn't work as expected
The `validation` command accepts additional args, where each one has alias (short one). So it can be run like:

```bash
monokle validate . --output sarif
monokle validate . -o sarif
```

However, `framework` `fw` alias is not parsed correctly:

```bash
monokle validate . --framework nsa # Works fine.
monokle validate . -fw nsa # Broken.
monokle validate . --fw nsa # Works fine.
```

Looks like this is the intended behavior of underlaying library, to allow single `-` aliases to be one letter only.

### PR DESCRIPTION
**Pull request by f1ames**
This PR introduces integration with remote policies, fixes https://github.com/kubeshop/monokle-saas/issues/1863 (see it for broader context too).

It also fixes #14 and fixes #4.

## Changes

- Introduced `whoami`, `login`, `logout`, `config`, `init` commands.
- Both `login` and `validate` commands have `--api-token` param which is designed to be used for M2M/CI/CD scenarios, so that interactive login flow can be omitted.
- Migrated to core packages - `@monokle/synchroznier` and `@monokle/parser`.
- Added basic test cc.
- Added CI job to run test/build/package for each PR to make sure nothing breaks.
- Updated README ([preview here](https://github.com/kubeshop/monokle-cli/blob/f1ames/feat/remote-policy-integration/README.md)).

## Fixes

- #4.
- #14.

## Testing

It can be run like:

```bash
npm run dev command...
npm run dev validate .
npm run dev login
# etc
```

## Checklist

- [x] tested locally
- [x] added new dependencies
- [x] updated the docs
- [x] added a test

### CODE REVIEW

**Comment by f1ames** [comment]
Rebased onto latest `master`.

File: README.md [hunk 1]
```diff
@@ -123,6 +171,65 @@ monokle validate --output sarif k8s-dir > results.sarif
...
+
+<p align="center">
+  <img src="docs/images/policies.png"/>
+</p>
+
+After the setup is done, you can run `monokle validate` command and it will use remote policy as long as you are logged in.
+
+## Using on CI/CD pipelines
+
+> We have dedicated [`Monokle GitHub Bot`](#monokle-github-bot) to integrate centralized policy management into GitHub Actions which gives more extensive integration.
```
Comment (WitoDelnat) [inline README.md]: You might have intended something else here? Below you correctly mention the Monokle GitHub Action in a comment but here you talk about the GH Bot wrt GitHub Actions?

File: src/commands/validate.io.ts [hunk 2]
```diff
@@ -1,10 +1,11 @@
 import {
-  getRuleForResult,
+  getRuleForResult, //@TODO migrate to v2 method
```
Comment (WitoDelnat) [inline src/commands/validate.io.ts]: You missed a TODO here

**Review by WitoDelnat — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (f1ames) [reply README.md]: Most probably 👍 I guess we should both mention GH Action and GH Bot here, right? GH Bot as being more "integrated" (with the rest of ecosystem) solution?

Comment (f1ames) [reply README.md]: Ahhhh, I know what I was trying to say, I used GitHub Action as a "generic" name describing GitHub CI/CD (because that's how [GH calls it](https://github.com/features/actions)). Meaning that you can use Monokle GH Bot to easily integrated validation into GitHub CI/CD. I'll try to reword it to make easier to understand.

File: src/commands/validate.io.ts (hunk 2 above)
Comment (f1ames) [reply src/commands/validate.io.ts]: Done 👍 I reworked the code to correctly support multiple runs from single output for V2 method (even though we don't really have any case yet where output will contain more than one AFAIR).

**Review by WitoDelnat — APPROVED** [review]
